src/main.rs

```diff
@@ -472,127 +498,137 @@ fn convert_to_minimap2_cigar(cigar: &str) -> String {
+            let op_start = start + op_start + 4;
+            if let Some(op_end) = debug_str[op_start..].find(",") {
+                let op_end = op_start + op_end;
+                let op = &debug_str[op_start..op_end];
+                
+                // Extract count
+                if let Some(cnt_start) = debug_str[op_end..].find("cnt: ") {
+                    let cnt_start = op_end + cnt_start + 5;
+                    if let Some(cnt_end) = debug_str[cnt_start..].find(" }") {
+                        let cnt_end = cnt_start + cnt_end;
+                        if let Ok(count) = debug_str[cnt_start..cnt_end].parse::<u32>() {
+                            let op_char = match op {
+                                "Match" => '=',
+                                "Sub" => 'X',
+                                "Ins" => 'I',
+                                "Del" => 'D',
+                                _ => '='
+                            };
+                            result.push_str(&format!("{}{}", count, op_char));
+                        }
+                    }
+                }
             }
         }
+        pos = start + 1;
     }
     
-    // Trim leading/trailing indels
-    let cigar = raw_cigar.chars()
-        .skip_while(|&c| c == 'D' || c == 'I')
-        .collect::<String>()
-        .trim_end_matches(|c| c == 'D' || c == 'I')
-        .to_string();
+    // If parsing failed, fall back to simple approach
+    if result.is_empty() {
+        result = "10=".to_string();
+    }
     
-    // Count matches and mismatches ignoring N positions in guide
-    let mut n_adjusted_mismatches = 0;
-    let mut matches = 0;
-    let mut gaps = 0;
-    let mut current_gap_size = 0;
-    let mut max_gap_size = 0;
-    let mut pos = 0;
+    result
+}
+
+fn scan_window_sassy(
+    guide: &[u8], 
+    window: &[u8], 
+    max_mismatches: u32, 
+    max_bulges: u32, 
+    max_bulge_size: u32,
+    min_match_fraction: f32, 
+    no_filter: bool
+) -> Option<(i32, String, u32, u32, u32, usize)> {
     
-    for c in cigar.chars() {
-        match c {
-            'X' => {
-                if pos < guide.len() && guide[pos] != b'N' {
-                    n_adjusted_mismatches += 1;
-                }
-                pos += 1;
-            },
-            'I' | 'D' => {
-                current_gap_size += 1;
-                if current_gap_size == 1 {
-                    gaps += 1;
-                }
-                max_gap_size = max_gap_size.max(current_gap_size);
-                if c == 'I' { pos += 1; }
-            },
-            'M' | '=' => {
-                current_gap_size = 0;
-                matches += 1;
-                pos += 1;
-            },
-            _ => ()
-        }
-    }
+    // Calculate maximum allowed errors
+    let max_errors = (max_mismatches + max_bulges) as usize;
+    
+    // Create SASSY searcher with DNA profile
+    let mut searcher: Searcher<Dna> = Searcher::new(false, None);
```

Comment:
Depending on how you use this, you may want to reuse the searcher object between consecutive invocations, so allocations are reused.